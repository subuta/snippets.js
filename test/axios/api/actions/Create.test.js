/* global expect, describe, it */

import { build, format, snippets as s } from 'bld.js'

import CreateAction from 'lib/axios/api/actions/Create'

import {
  Routes as RoutesConfig,
  Models as ModelsConfig
} from 'test/fixtures/config'

describe('axios/api/actions/Create', () => {
  it('should create axios Api create action with channel', () => {
    const channelRoute = RoutesConfig['channel']
    const channelModel = ModelsConfig['channel']

    expect(format(CreateAction(channelRoute, channelModel))).toMatchSnapshot()
  })

  it('should create axios Api create action with comment', () => {
    const channelRoute = RoutesConfig['comment']
    const channelModel = ModelsConfig['comment']

    expect(format(CreateAction(channelRoute, channelModel))).toMatchSnapshot()
  })
})
