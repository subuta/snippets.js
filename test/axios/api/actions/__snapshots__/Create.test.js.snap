// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`axios/api/actions/Create should create axios Api create action with channel 1`] = `
"export const create = (params) => {
  return request.post(\`/channels\`, {
    channel: params
  })
}
"
`;

exports[`axios/api/actions/Create should create axios Api create action with comment 1`] = `
"export const create = (params) => {
  return request.post(\`/channels/\${params.channelId}/comments\`, {
    comment: params
  })
}
"
`;
